src/nvim/CMakeLists.txt

```diff
@@ -953,3 +953,12 @@ add_custom_target(doc-eval DEPENDS ${GEN_EVAL_TOUCH})
 add_custom_target(doc-vim DEPENDS ${VIMDOC_FILES})
 add_custom_target(doc)
 add_dependencies(doc doc-vim doc-eval)
+
+add_custom_target(doctest
+  COMMAND $<TARGET_FILE:nvim_bin> -es --clean +'helptags ALL'
+  COMMAND $<TARGET_FILE:nvim_bin> -V1 -es --clean +\"lua require'scripts.gen_help_html'.test_validate\(\)\"
+  COMMAND $<TARGET_FILE:nvim_bin> -V1 -es --clean +\"lua require'scripts.gen_help_html'.test_gen\(\)\"
+  DEPENDS
+    doc
```

Comment:
@dundargoc I wasn't sure which build target this should depend on, so as to ensure that `build/runtime/doc/*.txt` is up-to-date and identical with the source (`./runtime/doc/*.txt`). On CI it doesn't matter much because a clean build will be performed; but when it comes to a local dev environment, changes on `./runtime/doc/*.txt` are NOT correctly reflected to `build/runtime/doc/*.txt` so the validation result could be not consistent. Maybe the previous functionaltest (`help_spec`) also had the same issue.

Reply:
This has been resolved:

- Confirmed that the previous `help_spec` test was reading `./build/runtime/doc`, which can be out of sync with the source doc files (`./runtime/doc`).
- Changed `lintdoc` job to *always* read and validate `./runtime/doc`; the `lintdoc.lua` script itself might read different helpdir if different `$VIMRUNTIME` is provided.
- Duplicate `build/runtime/doc` might need to be cleaned up in the future.

Reply:
@justinmk As a follow-up after this PR, we might be getting rid of (the duplicate) `build/runtime/doc` out of the build process. Do you see any unforseen problems here? (/cc @dundargoc)

Reply:
> As a follow-up after this PR, we might be getting rid of (the duplicate) `build/runtime/doc` out of the build process. Do you see any unforseen problems here?

No objection. I only was using it because it worked better than previous things I tried.